### ISSUE TITLE
change "fill" names to something more suggestive of the "not-yet-determined value" semantics

### ISSUE THREAD

**Issue by jeremy-w** [bignerdranch/Deferred#20]
"Fill" suggests a "box" viewpoint. This doesn't mesh well with the "not-yet-determined value" semantic POV.

Suggestions:

- determine
- resolve
- fix
- now
- settle

Splitting into Promise/Future means that we want a term more in line with the "promise" concept. Suggestions:

- resolve (used by the pasteboard APIs)
- fulfill (suggested by @lyricsboy and keeps the 'fill' feel without committing to a box concept)
- deliver

**Comment by zwaldowski** [comment]
I haven't found people hanging up on `fill`, though there's definitely an argument to be had. I feel `determine` is too wordy, and I don't _quite_ know why I was using it in the docs but for following the "not using the word in the definition" rule.

**Comment by lyricsboy** [comment]
I am fine with `fill` in my use, but `fulfill` might be a better word. It is basically a promise, after all.

**Comment by zwaldowski** [comment]
And in the case of #33 it's literally a `Promise`... `Type`. :P

**Comment by jeremy-w** [comment]
Can we make `Task` into `FingersCrossedPromise` so that you can `FingersCrossedPromise.break` rather than `Task.cancel`? ;)

`fulfill` plays well with the "Promise" verbiage for sure. Though I notice that the pasteboard API uses the term "resolve a promise" rather than "fulfill".

**Comment by zwaldowski** [comment]
`break` is a language keyword, so I think not. :wink:

**Comment by jeremy-w** [comment]
/me tableflips about Swift and its keyword-madness again.

**Comment by zwaldowski** [comment]
Then you might as well tableflip about C too...

**Comment by zwaldowski** [comment]
The most movement on this is `fulfill`. Do we want to commit to that?

**Comment by jeremy-w** [comment]
:-1: `fulfillIfUnfulfilled` is a beast to type. `resolve` is common and rolls off the tongue easier, but no-one seemed interested, so let's leave it as `fill`. It's used enough that brevity matters, too.

**Comment by zwaldowski** [comment]
`IfUnfulfilled` is no longer around after the Pod's decision. But alright!

**Comment by jeremy-w** [comment]
I just cruised through tests on master hunting for uses of `fill` to get a feel for what a more verbose word would do. Sounds like we really need to hit the merge button soon on a lot of things. :)

**Comment by zwaldowski** [comment]
Tests are now [a little more current](https://github.com/bignerdranch/Deferred/tree/master/DeferredTests). :grimacing:

**Comment by jeremy-w** [comment]
This is a bit jarring side-by-side:

```
            deferred.fill(42)
            expect.fulfill()
```

The "too much `ful`" worry lives on as `d.fill(2, assertIfFilled: false)` vs. `d.fulfill(2, assertIfFulfilled: false)`.

It occurs to me though that we could drop the "if" from that; it should be something like `assertUnfilled` (the claim we're asserting) or `assertSuccess` (vaguer as to what's succeeding, but avoids asserting a negative). Perhaps we could get away from the "assert" bit somehow - `d.fill(2, speculative: true)`, `d.fill(2, racing: true)`?

**Comment by zwaldowski** [comment]
Good thinking. However, I will note that the [primary version](https://github.com/bignerdranch/Deferred/blob/master/Deferred/PromiseType.swift#L36) doesn't involve any of this; the [asserting version](https://github.com/bignerdranch/Deferred/blob/master/Deferred/PromiseType.swift#L59-L63) was added back primarily to avoid churn in the tests.

**Comment by jeremy-w** [comment]
Do we to update the tests to the primary version and remove the asserting version?

**Comment by zwaldowski** [comment]
Not immediately. :grimacing: I left the asserting version in because some might find it useful.
